### PR TITLE
Adds custom configuration docs for Ingress Controller

### PR DESCRIPTION
**Pull request by puja108**
This should be ready for review now, would wait for the merge until we have it in an active release.

### CODE REVIEW

File: src/content/guides/advanced-ingress-configuration/index.md [hunk 1]
```diff
@@ -326,12 +326,42 @@ Check out the [ingress-nginx repository](https://github.com/kubernetes/ingress-n
...
-Your Giant Swarm installation may comes with some global defaults for Ingress Controller configuration. However, you can override these defaults by setting your per cluster configuration in form of a ConfigMap named `ingress-controller-config-custom` located in your `kube-system` namespace.
+Your Giant Swarm installation comes with some global defaults for Ingress Controller configuration. However, you can override these defaults by setting your per cluster configuration in form of a ConfigMap named `nginx-ingress-controller-user-values` located in your `kube-system` namespace.
+
+__Note__ that this feature is only available in more recent cluster versions. To check if your cluster version support customization of the ConfigMap, you can check if the above-mentioned ConfigMap is present.
+
+__Warning:__ Please do not edit any of the other nginx ingress related ConfigMaps. Only the user ConfigMap is safe to edit.
+
+```nohighlight
+kubectl -n kube-system get cm nginx-ingress-controller-user-values
+NAME                                   DATA      AGE
```
Comment (pipo02mix) [inline src/content/guides/advanced-ingress-configuration/index.md]: I think we use `$` to prefix commands

File: src/content/guides/advanced-ingress-configuration/index.md [hunk 2]
```diff
@@ -326,12 +326,42 @@ Check out the [ingress-nginx repository](https://github.com/kubernetes/ingress-n
 
 ## Global (per Cluster) Options {#configmap}
 
-Your Giant Swarm installation may comes with some global defaults for Ingress Controller configuration. However, you can override these defaults by setting your per cluster configuration in form of a ConfigMap named `ingress-controller-config-custom` located in your `kube-system` namespace.
+Your Giant Swarm installation comes with some global defaults for Ingress Controller configuration. However, you can override these defaults by setting your per cluster configuration in form of a ConfigMap named `nginx-ingress-controller-user-values` located in your `kube-system` namespace.
+
+__Note__ that this feature is only available in more recent cluster versions. To check if your cluster version support customization of the ConfigMap, you can check if the above-mentioned ConfigMap is present.
```
Comment (pipo02mix) [inline src/content/guides/advanced-ingress-configuration/index.md]: `support` -> `support s`

Comment (pipo02mix) [reply src/content/guides/advanced-ingress-configuration/index.md]: For me `the feature` sounds better than `that this feature`

**Comment by pipo02mix** [comment]
BTW the release is active :)

File: src/content/guides/advanced-ingress-configuration/index.md [hunk 3]
```diff
@@ -326,12 +326,42 @@ Check out the [ingress-nginx repository](https://github.com/kubernetes/ingress-n
...
+
+```nohighlight
+kubectl -n kube-system get cm nginx-ingress-controller-user-values
+NAME                                   DATA      AGE
+nginx-ingress-controller-user-values   0         11m
+```
 
 The official documentation of the NGINX Ingress Controller contains an overview of the [configuration options](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/) and their defaults.
 
+However, we currently only support a subset of those options, which we default like follows. If you need any other upstream documented option added to this list, please contact support.
```
Comment (rossf7) [inline src/content/guides/advanced-ingress-configuration/index.md]: as follows

**Comment by marians** [comment]
Checking now

File: src/content/guides/advanced-ingress-configuration/index.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 +++
 title = "Advanced Ingress Configuration"
 description = "Here we describe how you can customize and enable specific features for the NGINX-based Ingress"
-date = "2018-09-04"
+date = "2018-09-26"
```
Comment (marians) [inline src/content/guides/advanced-ingress-configuration/index.md]: The date will need another update before merging

File: src/content/guides/advanced-ingress-configuration/index.md [hunk 5]
```diff
@@ -326,12 +326,42 @@ Check out the [ingress-nginx repository](https://github.com/kubernetes/ingress-n
...
 
-Your Giant Swarm installation may comes with some global defaults for Ingress Controller configuration. However, you can override these defaults by setting your per cluster configuration in form of a ConfigMap named `ingress-controller-config-custom` located in your `kube-system` namespace.
+Your Giant Swarm installation comes with some global defaults for Ingress Controller configuration. However, you can override these defaults by setting your per cluster configuration in form of a ConfigMap named `nginx-ingress-controller-user-values` located in your `kube-system` namespace.
+
+__Note__ that this feature is only available in more recent cluster versions. To check if your cluster version support customization of the ConfigMap, you can check if the above-mentioned ConfigMap is present.
+
+__Warning:__ Please do not edit any of the other nginx ingress related ConfigMaps. Only the user ConfigMap is safe to edit.
+
+```nohighlight
+kubectl -n kube-system get cm nginx-ingress-controller-user-values
```
Comment (marians) [inline src/content/guides/advanced-ingress-configuration/index.md]: Please prepend `$` and a blank to the command

File: src/content/guides/advanced-ingress-configuration/index.md (hunk 3 above)
Comment (marians) [inline src/content/guides/advanced-ingress-configuration/index.md]: I'd remove `However`

File: src/content/guides/advanced-ingress-configuration/index.md (hunk 2 above)
Comment (marians) [inline src/content/guides/advanced-ingress-configuration/index.md]: `__Note__ that this feature` should be `__Note__: this feature`

File: src/content/guides/advanced-ingress-configuration/index.md [hunk 6]
```diff
@@ -326,12 +326,42 @@ Check out the [ingress-nginx repository](https://github.com/kubernetes/ingress-n
...
+  hsts: "false"
+  server-name-hash-bucket-size: "1024"
+  server-name-hash-max-size: "1024"
+  server-tokens: "false"
+  worker-processes: "4"
+  enable-underscores-in-headers: ""
+  proxy-buffers-size: ""
+  proxy-buffers: ""
+  vts-default-filter-key: ""
+```
```
Comment (marians) [inline src/content/guides/advanced-ingress-configuration/index.md]: I'd love a description what this code block is showing. Is it the defaults we use? If yes, will we keep this up to date here? Or is it an example?

File: src/content/guides/advanced-ingress-configuration/index.md [hunk 7]
```diff
@@ -326,12 +326,42 @@ Check out the [ingress-nginx repository](https://github.com/kubernetes/ingress-n
...
+
+On cluster creation the ConfigMap is empty and above defaults will be applied to the final Ingress Controller deployment. To override any of the above values, you just need to add the respctive line in the data field of the user ConfigMap.
```
Comment (marians) [inline src/content/guides/advanced-ingress-configuration/index.md]: OK. I missed this here. Maybe turn around the structure, to make clear that this paragraph explains the code block.

`The above defaults ...`.

**Comment by marians** [comment]
I'm done. Would like some improvements, but nothing blocking.

One thing that always appears weird to me is the capitalization of names like `NGINX Ingress Controller`. But that may be just taste.

File: src/content/guides/advanced-ingress-configuration/index.md (hunk 6 above)
Comment (marians) [reply src/content/guides/advanced-ingress-configuration/index.md]: Missed that. There you see how I review ;-)

**Pull request merged**